controllers/providers/aws/aws.go

```diff
@@ -187,6 +188,17 @@ func (w *AwsWorker) compactTags(tags []map[string]string) map[string]string {
 	return compacted
 }
 
+type AwsFargateWorker struct {
```

Comment:
Can you re-use the existing AwsWorker? we should be remvoing any state from this struct (including the EKSCF ones) - state should be kept in DiscoveredState (ProfileName, ClusterName, Selectors, Subnets, Tags).
Now that we've merged #83 - consider using existing methods around IAM if they can be reused.

Reply:
So take out derivable data from AwsFargateWorker  and then merge AwsWorker and AwsFargateWorker?

Reply:
Yes, but they are specific to the provisioner, hence should be part of the provisioner implementation.
I think it's better to have a single struct for AwsWorker which only holds the clients (and methods).

This needs to be done for the eks-managed provisioner as well.
We can also do this down the road if it's a big refactor

Reply:
OK.  When I add the AwsFargateWorker methods to AwsWorker, I'll add parameters on the methods for the fields that were once in the AwsFargateWorker struct.

Reply:
done.